src/shared/models/origin.ts

```diff
@@ -0,0 +1,60 @@
+import * as Rt from 'runtypes';
+
+export type LocalOrigin = {
+  type: 'local';
+  filePath: string;
+};
+export type ClusterOrigin = {
+  type: 'cluster';
+  context: string;
+};
+export type HelmOrigin = {
+  type: 'helm';
+  chartId: string;
+  valuesFilePath: string;
+};
+export type KustomizeOrigin = {
```

Comment:
should we add a "CommandOrigin" for resources that are generated by a preview command?

Reply:
good point, thanks! I'll add it